artifactory.py

```diff
@@ -891,19 +891,41 @@ def unlink(self, pathobj):
         Removes a file or folder
         """
 
+        if not pathobj.exists():
+            raise OSError(2, f"No such file or directory: {pathobj}")
+
         url = "/".join(
             [
                 pathobj.drive.rstrip("/"),
                 str(pathobj.relative_to(pathobj.drive)).strip("/"),
             ]
         )
-        self.rest_del(
-            url,
-            session=pathobj.session,
-            verify=pathobj.verify,
-            cert=pathobj.cert,
-            timeout=pathobj.timeout,
-        )
+
+        try:
+            self.rest_del(
+                url,
+                session=pathobj.session,
+                verify=pathobj.verify,
+                cert=pathobj.cert,
+                timeout=pathobj.timeout,
+            )
+        except requests.exceptions.HTTPError as err:
+            if err.response.status_code == 404:
+                # since we performed existence check we can say it is permissions issue
+                # see https://github.com/devopshq/artifactory/issues/36
+                errors = json.loads(err.response.text)["errors"][0]
```

Comment:
Isn't `err.response` a `request.Response` instance? If it's - we can use `err.response.json()`

Reply:
very good point, I will do a check if json is available straight away